src/graphics/images/dds.cpp

```diff
@@ -55,38 +64,56 @@ DDS::DDS(Common::ReadStream *dds) {
 	uint32_t pixelFormatFlags = dds->readUint32LE();
 	uint32_t fourCC = dds->readUint32BE();
 
-	switch (fourCC) {
-		case DDS_DXT5:
-			_format = kBC3;
-			_compressed = true;
-			break;
-		case DDS_DXT3:
-			_format = kBC2;
-			_compressed = true;
-			break;
-		case DDS_DXT1:
-			_format = kBC1;
-			_compressed = true;
-			break;
-		default:
-			throw std::runtime_error("Invalid pixel format");
-	}
-
 	uint32_t rgbBitCount = dds->readUint32LE();
 
 	uint32_t redBitMask   = dds->readUint32LE();
 	uint32_t greenBitMask = dds->readUint32LE();
 	uint32_t blueBitMask  = dds->readUint32LE();
 	uint32_t alphaBitMask = dds->readUint32LE();
 
+	uint32_t bytesPerPixel;
+	if (pixelFormatFlags & DDPF_FOURCC) {
+		switch (fourCC) {
+			case DDS_DXT5:
+				_format = kBC3;
+				_compressed = true;
+				break;
+			case DDS_DXT3:
+				_format = kBC2;
+				_compressed = true;
+				break;
+			case DDS_DXT1:
+				_format = kBC1;
+				_compressed = true;
+				break;
+			default:
+				throw std::runtime_error("Unsupported dds pixel format - Unknown fourCC Code");
+		}
+	} else if ((pixelFormatFlags & DDPF_RGBA)
+				&& (redBitMask   == 0x00FF0000)
+				&& (greenBitMask == 0x0000FF00)
+				&& (blueBitMask  == 0x000000FF)
+				&& (alphaBitMask == 0xFF000000)
+			  ) {
+		_format = kRGBA8;
+		_compressed = false;
+		bytesPerPixel = 4;
+	} else
+		throw std::runtime_error("Unsupported dds pixel format");
+
 	dds->skip(20);
 
 	_layers.resize(mipMapCount);
 	for (auto &mipMap: _layers) {
 		mipMap.width = width;
 		mipMap.height = height;
 
-		size_t dataSize = std::max(16u, ((width + 3u) / 4u) * ((height + 3u) / 4u) * 16u);
```

Comment:
You can get the image data size over the getImageSize from the base class and won't need to implement this here